manager/integration/tests/test_scheduling.py

```diff
@@ -454,6 +468,424 @@ def test_data_locality_basic():
+
+    volume1 = client.by_id_volume(volume1_name)
+
+    volume1 = wait_for_volume_replica_count(client, volume1_name, 1)
+    assert len(volume1.replicas) == 1
+    assert volume1.replicas[0]['hostId'] == volume1_attached_node
+
+    delete_and_wait_pod(core_api, pod1_name)
+    volume1 = wait_for_volume_detached(client, volume1_name)
+
+    volume1_replica_node = volume1.replicas[0]['hostId']
+
+    volume1_attached_node = None
+    for node in nodes:
+        if node.name != volume1_replica_node:
+            volume1_attached_node = node.name
+            break
+
+    assert volume1_attached_node is not None
+
+    pod1['spec']['nodeSelector'] = \
+        {"kubernetes.io/hostname": volume1_attached_node}
+    create_and_wait_pod(core_api, pod1)
+
+    wait_for_rebuild_start(client, volume1_name)
+
+    volume1 = client.by_id_volume(volume1_name)
+    assert len(volume1.replicas) == 2
+
+    wait_for_rebuild_complete(client, volume1_name)
+
+    volume1 = client.by_id_volume(volume1_name)
+    volume1 = wait_for_volume_replica_count(client, volume1_name, 1)
+    assert len(volume1.replicas) == 1
+    assert volume1.replicas[0]['hostId'] == volume1_attached_node
+
+    node1 = nodes[0]
+    node2 = nodes[1]
+    node3 = nodes[2]
+
+    client.update(node1, allowScheduling=True, tags=["AVAIL"])
+    client.update(node2, allowScheduling=True, tags=["AVAIL"])
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    try:
+        client.update(replica_node_soft_anti_affinity_setting,
+                      value="true")
+    except Exception as e:
+        print("Exception when update "
+              "Replica Node Level Soft Anti-Affinity setting",
+              replica_node_soft_anti_affinity_setting, e)
+
+    # case 2
+    volume2_name = volume_name + "-2"
+    volume2_size = str(500 * Mi)
+    pv2_name = volume2_name + "-pv"
+    pvc2_name = volume2_name + "-pvc"
+    pod2_name = volume2_name + "-pod"
+    pod2 = pod
+
+    pod2['metadata']['name'] = pod2_name
+
+    volume2 = client.create_volume(name=volume2_name,
+                                   size=volume2_size,
+                                   numberOfReplicas=3,
+                                   nodeSelector=["AVAIL"],
+                                   dataLocality="best-effort")
+
+    volume2 = wait_for_volume_detached(client, volume2_name)
+    volume2 = client.by_id_volume(volume2_name)
+    create_pv_for_volume(client, core_api, volume2, pv2_name)
+    create_pvc_for_volume(client, core_api, volume2, pvc2_name)
+
+    volume2 = client.by_id_volume(volume2_name)
+
+    pod2['spec']['volumes'] = [{
+        "name": "pod-data",
+        "persistentVolumeClaim": {
+            "claimName": pvc2_name
+        }
+    }]
+
+    pod2['spec']['nodeSelector'] = {"kubernetes.io/hostname": node3.name}
+    create_and_wait_pod(core_api, pod2)
+
+    volume2 = wait_for_volume_healthy(client, volume2_name)
+
```

Comment:
Can we check to make sure that there are 3 replicas at this moment?
` assert len(volume1.replicas) == 3`